File: src/main/java/org/blacksun/graph/channel/SimplexChannelFactory.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

public class SimplexChannelFactory implements ChannelFactory {
    @Override
    public Channel createChannel(@NotNull GraphNode from, @NotNull GraphNode to, int weight, double errors) {
        return new SimplexChannel(from, to, weight, errors);
    }
}

File: src/main/java/org/blacksun/graph/channel/SimplexChannelFactory.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode

class SimplexChannelFactory : ChannelFactory {
    override fun createChannel(from: GraphNode, to: GraphNode, weight: Int, errors: Double): Channel {
        return SimplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Simplex"
    }
}
